AdminUi/src/AdminUi/ClientApp/src/app/services/identity-service/identity.service.ts

```diff
@@ -11,23 +15,172 @@ import { Quota } from "../quotas-service/quotas.service";
+                    break;
+                case "<=":
+                    odataFilter.le("numberOfDevices", filter.numberOfDevices.value);
+                    break;
+                case ">=":
+                    odataFilter.ge("numberOfDevices", filter.numberOfDevices.value);
+                    break;
+                default:
+                    this.logger.error(`Invalid numberOfDevices filter operator: ${filter.numberOfDevices.operator}`);
+                    break;
+            }
+        }
+
+        if (filter.datawalletVersion.operator !== undefined && filter.datawalletVersion.value?.valueOf) {
+            switch (filter.datawalletVersion.operator) {
+                case ">":
+                    odataFilter.gt("datawalletVersion", filter.datawalletVersion.value);
+                    break;
+                case "<":
+                    odataFilter.lt("datawalletVersion", filter.datawalletVersion.value);
+                    break;
+                case "=":
+                    odataFilter.eq("datawalletVersion", filter.datawalletVersion.value);
+                    break;
+                case "<=":
+                    odataFilter.le("datawalletVersion", filter.datawalletVersion.value);
+                    break;
+                case ">=":
+                    odataFilter.ge("datawalletVersion", filter.datawalletVersion.value);
+                    break;
+                default:
+                    this.logger.error(`Invalid datawalletVersion filter operator: ${filter.datawalletVersion.operator}`);
+                    break;
+            }
+        }
+
+        if (filter.identityVersion.operator !== undefined && filter.identityVersion.value?.valueOf) {
+            switch (filter.identityVersion.operator) {
+                case ">":
+                    odataFilter.gt("identityVersion", filter.identityVersion.value);
+                    break;
+                case "<":
+                    odataFilter.lt("identityVersion", filter.identityVersion.value);
+                    break;
+                case "=":
+                    odataFilter.eq("identityVersion", filter.identityVersion.value);
+                    break;
+                case "<=":
+                    odataFilter.le("identityVersion", filter.identityVersion.value);
+                    break;
+                case ">=":
+                    odataFilter.ge("identityVersion", filter.identityVersion.value);
+                    break;
+                default:
+                    this.logger.error(`Invalid identityVersion filter operator: ${filter.identityVersion.operator}`);
+                    break;
+            }
+        }
+
+        let filterParameter = "";
+        if (odataFilter.toString() !== "") filterParameter = `?$filter=${odataFilter.toString()}`;
+        if (filterParameter === "") {
+            filterParameter += `?${paginationFilter}`;
+        } else {
+            filterParameter += `&${paginationFilter}`;
+        }
```

Comment:
Wouldn't this work:

```typescript
const filterComponents = [odataFilter.toString(), paginationFilter];
const filterParameter = `?${filterComponents.join("&")}`;
```
Since the `paginationFilter` seems to always be set, this should work, and would be way easier to understand imo.

Reply:
that approach is missing the $filter anottation based on whether we are applying a filter or just the pagination.